p2p_wallet/Services/Storage/KeychainStorage.swift

```diff
@@ -105,9 +105,25 @@ class KeychainStorage {
             UserDefaults.standard.set(true, forKey: ubiquitousKeyValueStoreToKeychain)
         }
 
-        [pincodeKey, phrasesKey, derivableTypeKey, walletIndexKey].forEach { key in
-            guard let data = icloudKeychain.getData(key), localKeychain.getData(key) == nil else { return }
-            localKeychain.set(data, forKey: key)
+        // migrate from iCloud keychain to localKeychain
+        let iCloudKeychainToLocalKeychainMigrated = "iCloudKeychainToLocalKeychainMigrated"
+        if !UserDefaults.standard.bool(forKey: iCloudKeychainToLocalKeychainMigrated) {
+            // safely check all keys in localKeychain, only override when data in localKeychain is empty
+            if localKeychain.getData(pincodeKey) == nil,
+               localKeychain.getData(pincodeAttemptsKey) == nil,
```

Comment:
i guess there is no need in pincodeAttemptsKey and ethAddressKey here, because they are new for this version and might not need migration later? we don't have these keys in develop

Reply:
Yes, But for safety, we have to check it all, to make sure that no data is overiden

Reply:
got it